Update Elasticsearch recipe with correct download URLs

radamanthus opened the pull request.

crigor commented:
What is the value of `#{Chef::Config[:file_cache_path]`? Consider putting it on /tmp or somewhere we don't delete. I think we delete /etc/chef/recipes or /etc/chef/recipes/cookbooks/. Also add `checksum` so the file isn't downloaded on every chef run.

radamanthus commented:
`#{Chef::Config[:file_cache_path]` is `/etc/chef/recipes`. I replaced references to that with `/mnt/tmp` (configurable via attributes) and added `checksum`.

crigor commented:
LGTM
